src/routes/categories/containers/__tests__/CategoriesPage.spec.js

```diff
@@ -292,89 +291,76 @@ describe('CategoriesPage', () => {
 
-    const languagesEndpoint = new EndpointBuilder('languages')
+    const categoriesEndpoint = new EndpointBuilder('categories')
       .withUrl('https://weird-endpoint/api.json')
       .withMapper(json => json)
-      .withResponseOverride(languages)
+      .withResponseOverride(categories)
       .build()
 
+    const pathname = '/augsburg/de/willkommen'
+    const id = '1234'
+
     test('should map state to props', () => {
-      const pathname = '/augsburg/de'
-      const id = 1234
-      const store = mockStore({
-        categories: new Payload(false),
-        languages: new Payload(false),
-        locations: new Payload(false),
-        router: {params: {location: location, language: language}, pathname: pathname, query: {id: id}}
+      const store = createReduxStore(createHistory, {
+        router: {params: {location: location, language: language},
+          pathname: pathname,
+          query: {id: id}},
+        languageChangeUrls: {}
       })
 
-      const tree = mount(
+      const categoriesPage = mount(
         <Provider store={store}>
           <EndpointProvider endpoints={[categoriesEndpoint, locationsEndpoint, languagesEndpoint]}>
             <ConnectedCategoriesPage />
           </EndpointProvider>
         </Provider>
-      )
-
-      const categoriesPageProps = tree.find(ConnectedCategoriesPage).childAt(0).props()
+      ).find(CategoriesPage)
 
-      // todo add categories, locations and languages
-      expect(categoriesPageProps).toEqual({
+      expect(categoriesPage.props()).toEqual({
         location: location,
         language: language,
         path: pathname,
         categoryId: id,
         setLanguageChangeUrls: expect.any(Function),
-        replaceUrl: expect.any(Function)
+        replaceUrl: expect.any(Function),
+        categories: categories,
+        locations: locations,
+        languages: languages
       })
     })
 
-    // we don't have to test replaceUrl here, because it will be redundant after availableLanguages
-    // in the cms are changed to the path instead of the id
     test('should map dispatch to props', () => {
-      const pathname = '/augsburg/de'
-      const store = mockStore({
-        categories: new Payload(false),
-        languages: new Payload(false),
-        locations: new Payload(false),
-        router: {params: {location: location, language: language}, pathname: pathname, query: {}}
+      const store = createReduxStore(createHistory, {
+        router: {params: {location: location, language: language},
+          pathname: pathname,
+          query: {id: id}},
+        languageChangeUrls: {}
       })
 
-      const mapLanguageToUrl = (language, id) => 'test' + language + id
-
-      const testUrls = {
-        en: 'testenundefined',
-        de: 'testde1235',
-        ar: 'testar1236'
+      const mapLanguageToUrl = (language, id) => `/${language}/${id}`
+      const languageChangeUrls = {
+        en: '/en/1235',
+        de: '/de/undefined',
```

Comment:
Should we maybe exclude this the de key if it's undefined?  Looks weird if we test for that invalid url.

Reply:
fixed